test/e2e/setup/setup.go

```diff
@@ -112,7 +114,18 @@ func GeneratePasswordForUser(mdbu mdbv1.MongoDBUser, ctx *e2eutil.Context, names
 		SetField(passwordKey, password).
 		Build()
 
-	return password, e2eutil.TestClient.Create(context.TODO(), &passwordSecret, &e2eutil.CleanupOptions{TestContext: ctx})
+	err = e2eutil.TestClient.Create(context.TODO(), &passwordSecret, &e2eutil.CleanupOptions{TestContext: ctx})
+
+	// as we just need there to be a password, it's fine to re-use an existing one (from a previous test run if it was not deleted).
+	if apiErrors.IsAlreadyExists(err) {
+		existingSecret := corev1.Secret{}
+		if err := e2eutil.TestClient.Get(context.TODO(), types.NamespacedName{Name: mdbu.PasswordSecretRef.Name, Namespace: namespace}, &existingSecret); err != nil {
```

Comment:
should we use `nps` instead of `namespace` when building the NamespacedName since this is used when building the passwordSecret (line 111)?

Reply:
good catch yes we should.